epochStart/metachain/epochStartData.go

```diff
@@ -205,6 +207,10 @@ func (e *epochStartData) lastFinalizedFirstPendingListHeadersForShard(shardHdr *
 		prevShardHdr, err := process.GetShardHeader(currentHdr.GetPrevHash(), e.dataPool.Headers(), e.marshalizer, e.store)
 		if err != nil {
 			go e.requestHandler.RequestShardHeader(currentHdr.ShardID, currentHdr.GetPrevHash())
+			if e.epochStartTrigger.Epoch()-currentHdr.GetEpoch() > maxEpochDifference {
+				log.Warn("shard remained in too old epoch", "shardID", currentHdr.ShardID, "shard Epoch", currentHdr.Epoch, "meta Epoch", e.epochStartTrigger.Epoch())
```

Comment:
"shard remained in an epoch that is too old"

Reply:
done.